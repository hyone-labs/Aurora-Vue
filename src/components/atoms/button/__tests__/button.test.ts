/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { render, screen, fireEvent } from "@testing-library/vue";
import { describe, it, expect, vi } from "vitest";
import HButton from "../button.vue";

describe("HButton", () => {
  it("render correctly with default props", () => {
    const { html } = render(HButton, {
      slots: {
        default: "My first test",
      },
    });
    expect(html()).toMatchSnapshot();
  });

  it("render correctly with different props", () => {
    const { html } = render(HButton, {
      slots: {
        default: "With many Itens",
      },
      props: {
        behavior: "block",
        color: "secondary",
        size: "small",
        iconLeft: "fas fa-circle",
        iconColor: "primary",
        isRounded: false,
      },
    });
    expect(html()).toMatchSnapshot();
  });

  it("render correctly with attrs", () => {
    const { html } = render(HButton, {
      slots: {
        default: "With many Itens",
      },
      attrs: {
        id: "my-first-component",
        "data-test-id": "secondary",
      },
    });
    expect(html()).toMatchSnapshot();
  });

  it("when click active event", async () => {
    const handleClick = vi.fn();
    render(HButton, {
      slots: {
        default: "Event",
      },
      attrs: {
        onClick: handleClick,
      },
    });

    await fireEvent.click(screen.getByText("Event").closest("button")!);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
