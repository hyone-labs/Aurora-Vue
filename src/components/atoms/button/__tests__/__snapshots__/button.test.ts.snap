// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HButton > render correctly with attrs 1`] = `
"<button class="h-button h-button--color-default h-button--size-small h-button--variant-contained" id="my-first-component" data-test-id="secondary">
  <!--v-if-->
  <!--v-if--><span class="h-button__text">With many Itens</span>
  <!--v-if-->
</button>"
`;

exports[`HButton > render correctly with default props 1`] = `
"<button class="h-button h-button--color-default h-button--size-small h-button--variant-contained">
  <!--v-if-->
  <!--v-if--><span class="h-button__text">My first test</span>
  <!--v-if-->
</button>"
`;

exports[`HButton > render correctly with different props 1`] = `
"<button class="h-button h-button--color-secondary h-button--size-small h-button--variant-contained" behavior="block" iconcolor="primary" isrounded="false">
  <div class="h-icon h-icon--size-medium h-button__icon h-button__icon--position-left"><i class="fas fa-circle fa-fw"></i></div>
  <!--v-if--><span class="h-button__text">With many Itens</span>
  <!--v-if-->
</button>"
`;
